AsyncDisplayKit/Details/ASPhotosFrameworkImageRequest.m

```diff
@@ -0,0 +1,169 @@
+//
+//  ASPhotosFrameworkImageRequest.m
+//  AsyncDisplayKit
+//
+//  Created by Adlai Holler on 9/25/15.
+//  Copyright © 2015 Facebook. All rights reserved.
+//
+
+#import "ASPhotosFrameworkImageRequest.h"
+#import "ASBaseDefines.h"
+
+NSString *const ASPhotosURLScheme = @"ph";
+
+static NSString *const _ASPhotosURLQueryKeyWidth = @"width";
+static NSString *const _ASPhotosURLQueryKeyHeight = @"height";
+
+// value is PHImageContentMode value
+static NSString *const _ASPhotosURLQueryKeyContentMode = @"contentmode";
+
+// value is PHImageRequestOptionsResizeMode value
+static NSString *const _ASPhotosURLQueryKeyResizeMode = @"resizemode";
+
+// value is PHImageRequestOptionsDeliveryMode value
+static NSString *const _ASPhotosURLQueryKeyDeliveryMode = @"deliverymode";
+
+// value is PHImageRequestOptionsVersion value
+static NSString *const _ASPhotosURLQueryKeyVersion = @"version";
+
+// value is 0 or 1
+static NSString *const _ASPhotosURLQueryKeyAllowNetworkAccess = @"network";
+
+static NSString *const _ASPhotosURLQueryKeyCropOriginX = @"crop_x";
+static NSString *const _ASPhotosURLQueryKeyCropOriginY = @"crop_y";
+static NSString *const _ASPhotosURLQueryKeyCropWidth = @"crop_w";
+static NSString *const _ASPhotosURLQueryKeyCropHeight = @"crop_h";
+
+@implementation ASPhotosFrameworkImageRequest
```

Comment:
All this is slightly complicated by the fact that the old AssetsLibrary used URLs to identify assets which is NOT what we're doing here. Here were identifying the asset and specifying all the options to request a given format